### PR TITLE
msp modules

### PR DESCRIPTION
**Pull request by shunj-nb**

### CODE REVIEW

File: hooks/hookexecution/executor.go [hunk 1]
```diff
@@ -160,7 +160,7 @@ func (e *hookExecutor) ExecuteProcessedAuctionStage(request *openrtb_ext.Request
 
 	stageName := hooks.StageProcessedAuctionRequest.String()
 	executionCtx := e.newContext(stageName)
-	payload := hookstage.ProcessedAuctionRequestPayload{BidRequest: request.BidRequest}
+	payload := hookstage.ProcessedAuctionRequestPayload{BidRequest: request.BidRequest, BidRequestWrapper: request}
```
Comment (Peiling-Ding) [inline hooks/hookexecution/executor.go:163]: Why do you need to pass both the Request and the Wrapper?

Comment (shunj-nb) [reply hooks/hookexecution/executor.go:163]: to avoid changing codes for other unrelated modules(openrtb.blocking)

File: exchange/exchange.go [hunk 2]
```diff
@@ -348,13 +348,12 @@ func (e *exchange) HoldAuction(ctx context.Context, r *AuctionRequest, debugLog
 		liveAdapters []openrtb_ext.BidderName
 	)
 
-	if len(r.StoredAuctionResponses) > 0 {
-		adapterBids, fledge, liveAdapters, err = buildStoredAuctionResponse(r.StoredAuctionResponses)
+	shouldMspBackfillBids := mspUpdateStoredAuctionResponse(r)
+	if len(r.StoredAuctionResponses) > 0 && !shouldMspBackfillBids {
+		adapterBids, fledge, liveAdapters, err, anyBidsReturned = mspApplyStoredAuctionResponse(r)
```
Comment (Peiling-Ding) [inline exchange/exchange.go:353]: For in-house ads feature, the best way to do it is setting up an endpoint and creating a bid adapter. If you think this way is over complicated for now, then `ExecuteAuctionResponseStage` is an alternative approach. We should avoid updating Prebid-server code as much as possible.

Comment (shunj-nb) [reply exchange/exchange.go:353]: we need to pass more parameters for ExecuteAuctionResponseStage approach and reprocess line 384 to 505

**Pull request merged**